src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -162,6 +163,35 @@ public void setExampleSubmission(Boolean exampleSubmission) {
         this.exampleSubmission = exampleSubmission;
     }
 
+    /**
+     * Removes sensitive information (e.g. example solution of the exercise) from the submission based on the role of the current user. This should be called before sending a
+     * submission to the client. IMPORTANT: Do not call this method from a transactional context as this would remove the sensitive information also from the entities in the
+     * database without explicitly saving them.
+     *
+     * @param authCheckService authorization check service to check what details can be hidden
+     * @param user current user
+     */
+    public void hideDetails(AuthorizationCheckService authCheckService, User user) {
```

Comment:
Yes you are right here. The problem for me was using it in the programming submission. I didn't want to have it in the `SubmissionService` since then `ProgrammingSubmissionService` should inherit it and I would have to refactor it as well. It makes perfect sense now to have it in `GenericSubmissionRepository`. :+1: